### PR TITLE
[core] Remove Constrain mode options

### PR DESCRIPTION
**Pull request by nagineni**
setLatLngBounds() API allows the map to be constrained to certain Lat/Lng limits, so there is no need of having Constrain mode options on Map to constrain the map boundaries.

### CODE REVIEW

File: platform/ios/src/MGLMapView.mm [hunk 1]
```diff
@@ -1230,11 +1230,6 @@ - (void)validateDisplayLink
     BOOL isVisible = self.superview && self.window;
     if (isVisible && ! _displayLink)
     {
-        if (_mbglMap && self.mbglMap.getConstrainMode() == mbgl::ConstrainMode::None)
-        {
-            self.mbglMap.setConstrainMode(mbgl::ConstrainMode::HeightOnly);
-        }
-
```
Comment (anderco) [inline platform/ios/src/MGLMapView.mm]: I think removing this will cause the issue fixed in #3887 to regress.

Comment (brunoabinader) [reply platform/ios/src/MGLMapView.mm]: @1ec5 is that still an issue? IMO If yes, I think that we could initialize the Map object with a fake size of e.g. 64x64 and that should be enough, no?

Comment (brunoabinader) [reply platform/ios/src/MGLMapView.mm]: /cc @julianrex

**Comment by tmpsantos** [comment]
+15 −162  == great

File: platform/ios/src/MGLMapView.mm (hunk 1 above)
Comment (julianrex) [reply platform/ios/src/MGLMapView.mm]: > I think removing this will cause the issue fixed in #3887 to regress.

Looks like it does.

**Comment by julianrex** [comment]
Added the `do not merge` label; I'll look at the iOS part shortly - and will issue a PR to this branch.

File: platform/ios/src/MGLMapView.mm (hunk 1 above)
Comment (1ec5) [reply platform/ios/src/MGLMapView.mm]: Per #3887, we don’t have control over the order in which the camera parameters are individually applied, nor over when they’re applied. If we did have control over that, I agree that a “pending coordinate” pattern would be more appropriate.

> On iOS, the only guarantee we have timing-wise is that all the inspectables are applied after initialization but before the view is added to a window.

Comment (1ec5) [reply platform/ios/src/MGLMapView.mm]: Per #3887, we don’t have control over the order in which the camera parameters are individually applied, nor over when they’re applied. If we did have control over that, I agree that a “pending coordinate” pattern would be more appropriate.

> On iOS, the only guarantee we have timing-wise is that all the inspectables are applied after initialization but before the view is added to a window.

**Comment by julianrex** [comment]
@nagineni would you mind rebasing this? There's been a lot of refactoring since this PR was created...

**Comment by julianrex** [comment]
Not closed. Correct @nagineni?

**Comment by julianrex** [comment]
Is this still valid @nagineni ?

**Pull request closed without merging**